### PR TITLE
server: Fix the OCI process arguments build routine

### PR DESCRIPTION
**Pull request by sameo**
We need to support a 2x2 matrix of use cases with both
kubelet giving us (command, args) slices and the OCI
image config file giving us (ENTRYPOINT, CMD) slices.

Here we always prioritize the kubelet information over
the OCI image one, and use the latter when the former
is incomplete.

Not that this routine will be slightly simpler when
issue #395 is fixed.

Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>

### CODE REVIEW

File: server/container_create.go [hunk 1]
```diff
@@ -86,35 +87,67 @@ func buildOCIProcessArgs(containerKubeConfig *pb.ContainerConfig, imageOCIConfig
...
+		// Here the kubelet command slice is empty.
+		if imageOCIConfig != nil {
+			// If the OCI image config has an ENTRYPOINT we
+			// use it as our process command.
+			// Otherwise we use the CMD slice if it's not
+			// empty.
+			if imageOCIConfig.Config.Entrypoint != nil {
+				processEntryPoint = imageOCIConfig.Config.Entrypoint
+			} else if imageOCIConfig.Config.Cmd != nil {
+				processEntryPoint = imageOCIConfig.Config.Cmd
```
Comment (mrunalp) [inline server/container_create.go]: Is this the behavior in k8s with docker today?

Comment (sameo) [inline server/container_create.go]: It seems to be, yes.

**Comment by mrunalp** [comment]
LGTM

**Comment by runcom** [comment]
LGTM

**Pull request merged**